### DISCUSSION TITLE
Google Map and "flip"

### DISCUSSION THREAD

**Discussion by simplenotezy** [floating-ui/floating-ui#1290] [Q&A]
I am using popperjs inside a Google Maps container, on which I have several markers.

When you click a marker, I want to ensure the marker is within view, and alignment is automatically set to whatever placement that fits within the map container.

However, I don't get accurate results. I have tried with:

```
  const { styles, attributes } = usePopper(referenceElement, popperElement, {
    modifiers: [
      { name: 'arrow', options: { element: arrowElement } },
      {
        name: 'flip',
        options: {
          boundary: mapInstance.getDiv(), // this is a reference to the Google Map Instance, and getDiv() returns the container
          padding: 200
        },
      },
    ],
    placement: 'top',
  });
```

However, it seems to vary what placement it choses. Before I dig any deeper, I'm curious if anybody else had been struggling with this, and perhaps knew a good solution?

Here is an example of the weird behavior. On first click, it has the correct placement, on the second click (same marker) it is for some reason "top".

![google map example](https://user-images.githubusercontent.com/3786627/115961235-d3b9d680-a515-11eb-94a1-074b443146cf.gif)

**Comment by FezVrasta** [comment]
Are you sure the tooltip content is not changing after popper positions it?

**Reply by simplenotezy** [reply]
@FezVrasta Yes, the content is exactly the same. Styling is still WIP, so perhaps there is some styling that I could apply now with benefits? E.g. fixed width/height?

**Reply by FezVrasta** [reply]
Without a repro I'm unable to help further
